addons/im_livechat/static/src/embed/core/livechat_service.js

```diff
@@ -153,13 +153,13 @@ export class LivechatService {
         } finally {
             this.persistThreadPromise = null;
         }
-        const chatWindow = this.store.ChatWindow.records.find(
+        const chatWindow = this.store.discuss.chatWindows.find(
             (c) => c.thread.id === this.TEMPORARY_ID
         );
         if (chatWindow) {
-            this.env.services["mail.thread"].remove(chatWindow.thread);
+            chatWindow.thread?.delete();
             if (!this.thread) {
-                this.chatWindowService.close(chatWindow);
+                await this.chatWindowService.close(chatWindow);
```

Comment:
note: closing chat window on thread deletion should be automatically done in base code instead (is causual/on delete cascade equivalent)